frontend/src/stimulus/controllers/beforeunload.controller.ts

```diff
@@ -0,0 +1,75 @@
+import { ApplicationController } from 'stimulus-use';
+import { TurboBeforeVisitEvent } from '@hotwired/turbo';
+
+export function setPageWasEdited() {
+  const application = window.Stimulus;
+  const controller = application.getControllerForElementAndIdentifier(document.body, 'beforeunload') as BeforeunloadController;
+  if (controller) {
+    controller.pageWasEdited = true;
+  }
+}
+
+export function setPageWasSubmitted() {
+  const application = window.Stimulus;
+  const controller = application.getControllerForElementAndIdentifier(document.body, 'beforeunload') as BeforeunloadController;
+  if (controller) {
+    controller.pageWasEdited = true;
+  }
+}
+
+export class BeforeunloadController extends ApplicationController {
+  /** Globally setable variable whether the page was edited */
+  public pageWasEdited = false;
+
+  /** Globally setable variable whether the page form is submitted.
+   * Necessary to avoid a data loss warning on beforeunload */
+  public pageIsSubmitted = false;
```

Comment:
Perhaps this could be modelled with a single state property instead? (since source of truth)

```suggestion
  pageState:'loaded' | 'edited' | 'submitted' = 'loaded';
```

We could still provide accessor methods:

```ts
  public set wasLoaded():void {
    this.pageState = 'loaded';
  }

  public set wasEdited():void {
    this.pageState = 'edited';
  }

  public set wasSubmitted():void {
    this.pageState = 'submitted';
  }
```